File: lab-1/domain/individual_album_manager.py
from content_manager import ContentManager


class IndividualAlbumManager(ContentManager):
    def __init__(self, name):
        super().__init__(name)
        self.album = None  # A single album managed

    def set_album(self, album):
        self.album = album

    def add_item(self, item):
        if self.album:
            self.album.add_item(item)

    def remove_item(self, item):
        if self.album:
            self.album.remove_item(item)

    def display_contents(self):
        if self.album:
            self.album.display_info()

File: src/main/kotlin/content_manager/implementations/IndividualAlbumManager.kt
package content_manager.implementations
import content_manager.ContentManager
import media_item.MediaItem
import media_item.implementations.Album
import items_manager.implementations.AlbumItemsManager

class IndividualAlbumManager(override val name: String) : ContentManager {
    var album: Album? = null
    var albumItemsManager: AlbumItemsManager? = null

    fun setAlbum(album: Album) {
        this.album = album
        this.albumItemsManager = AlbumItemsManager(album)
    }

    override fun addItem(item: MediaItem) {
        if (album != null) {
            albumItemsManager?.addItem(item)
        }
    }

    override fun removeItem(item: MediaItem) {
        if (album != null) {
            albumItemsManager?.removeItem(item)
        }
    }

    override fun displayContents() {
        album?.displayInfo()
    }
}